app/src/main/java/com/tristanwiley/chatse/chat/ChatActivity.kt

```diff
@@ -450,6 +456,20 @@ class ChatActivity : AppCompatActivity(), ServiceConnection {
         Log.d("ChatActivity", "Service disconnect")
     }
 
+    override fun onItemClick(chatRoom: ChatRoom) {
+
+        val roomNo = chatRoom.num
+        if ((roomNum!=-1) && roomNum != roomNo) {
```

Comment:
This block is confusing because there are 2 variables that have almost the same name. Give them distinctive names like `currentRoomNr` and `clickedRoomNr` or something.